lib/mapObjects/CObjectHandler.h

```diff
@@ -114,26 +112,25 @@ class DLL_LINKAGE CGObjectInstance : public IObjectInterface
 {
 public:
 	/// Position of bottom-right corner of object on map
-	int3 pos;
+	int3 pos = {-1, -1, -1};
```

Comment:
To delete a constructor, as before. Do we need to save one and only constructor, if it defines only placeholders? What approach should be used then?

Reply:
Basically same as discussion below:
https://github.com/vcmi/vcmi/pull/1567#discussion_r1103700205
Except that this is about CGObjectInstance, base class for half of library code. Which IMO should be kept as minimal as possible and as clear as possible.

Reply:
@IvanSavenko I think this is better to init int3's default constructor to -1,-1,-1 rather than 0,0,0.
What do you think about it?

Reply:
Hmm...
Potential problem is that there may be code that relies on having current (0,0,0) as default value. Might be difficult to track every such case.

As for using (-1,-1,-1) as default constructor - if goal is to have "clearly invalid" value as default then something even less "normal" might be a better idea, e.g. `std::numeric_limits::min/max`.

Reply:
@IvanSavenko there is a comment in int3.h:

```c++
	si32 x, y, z;

	//c-tor: x, y, z initialized to 0
	int3() : x(0), y(0), z(0) {} // I think that x, y, z should be left uninitialized.
```

So, we can just declare default initializers for x, y, z as clearly invalid (either -1 or min/max). What value do you prefer for?

Reply:
That comment is ancient. I won't rely on it without some checking to make sure that there is no code that uses default values. For example, by removing/making private this constructor and check what fails to compile.

Reply:
int3 default constructor is called anywhere (likely in each file of lib/*). It seems almost impossible to track if code rely on fact than default values is correct, I think.

Reply:
Hmm... perhaps we can try tricking compiler/clang-tidy into telling that? By NOT initializing values in constructor? Should in theory lead to "maybe uninitialized" warning where needed.

Anyway, I am not against better tracking of uninitialized values, but this needs to be well-tested to minimize chance of regressions. If compiler can't do this part it will have to be done manually.

Reply:
Reverted change, but set TODO.